File: java/Samples/BiometricStandards/CS/ANTemplateSampleCS/RecordCreateForms/ANType14RecordCreateForm.cs
using System;
using Neurotec.Biometrics.Standards;
using Neurotec.IO;
using Neurotec.Images;

namespace Neurotec.Samples.RecordCreateForms
{
	public partial class ANType14RecordCreateForm : ANRecordCreateForm
	{
		#region Public constructor

		public ANType14RecordCreateForm()
		{
			InitializeComponent();
		}

		#endregion

		#region Protected methods

		protected override ANRecord OnCreateRecord(ANTemplate template)
		{
			ANType14Record record;
			if (imageLoader.CreateFromImage)
			{
				record = new ANType14Record(ANTemplate.VersionCurrent, Idc, imageLoader.Src, imageLoader.ScaleUnits,
					imageLoader.CompressionAlgorithm, imageLoader.Image);
			}
			else if (imageLoader.CreateFromData)
			{
				using (NImage image = NImage.FromMemory(imageLoader.ImageData))
				{
					record = new ANType14Record(ANTemplate.VersionCurrent, Idc, imageLoader.Src, imageLoader.ScaleUnits,
						imageLoader.CompressionAlgorithm, image)
						{
							HorzLineLength = imageLoader.Hll,
							VertLineLength = imageLoader.Vll,
							HorzPixelScale = imageLoader.Hps,
							VertPixelScale = imageLoader.Vps,
							BitsPerPixel = imageLoader.Bpx
						};
				}
			}
			else
			{
				throw new NotSupportedException();
			}

			template.Records.Add(record);
			return record;
		}

		#endregion
	}
}

File: java/Samples/BiometricStandards/VB/ANTemplateSampleVB/RecordCreateForms/ANType14RecordCreateForm.vb
Imports System
Imports Neurotec.Biometrics.Standards
Imports Neurotec.IO
Imports Neurotec.Images

Namespace RecordCreateForms

	Partial Public Class ANType14RecordCreateForm
		Inherits ANRecordCreateForm
#Region "Public constructor"

		Public Sub New()
			InitializeComponent()
		End Sub

#End Region

#Region "Protected methods"

		Protected Overrides Function OnCreateRecord(ByVal template As ANTemplate) As ANRecord
			Dim record As ANType14Record
			If imageLoader.CreateFromImage Then
				record = New ANType14Record(ANTemplate.VersionCurrent, Idc, imageLoader.Src, imageLoader.ScaleUnits, imageLoader.CompressionAlgorithm, imageLoader.Image)
			ElseIf imageLoader.CreateFromData Then
				Using image As NImage = NImage.FromMemory(imageLoader.ImageData)
					record = New ANType14Record(ANTemplate.VersionCurrent, Idc, imageLoader.Src, imageLoader.ScaleUnits, imageLoader.CompressionAlgorithm, image) With {.HorzLineLength = imageLoader.Hll, .VertLineLength = imageLoader.Vll, .HorzPixelScale = imageLoader.Hps, .VertPixelScale = imageLoader.Vps, .BitsPerPixel = imageLoader.Bpx}
				End Using
			Else
				Throw New NotSupportedException()
			End If

			template.Records.Add(record)
			Return record
		End Function

#End Region
	End Class
End Namespace
